rct229/rulesets/ashrae9012019/section5/section5rule41.py

```diff
@@ -0,0 +1,130 @@
+            "proposed design. ",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-5 Building Envelope Modeling Requirements for the Baseline building",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0].buildings[*]",
+            data_items={"climate_zone": (BASELINE_0, "weather/climate_zone")},
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule41.BuildingRule, self).__init__(
+                rmrs_used=produce_ruleset_model_instance(
+                    USER=False, BASELINE_0=True, PROPOSED=True
+                ),
+                each_rule=Section5Rule41.BuildingRule.RoofRule(),
+                index_rmr=BASELINE_0,
+                list_path="$.building_segments[*].zones[*].surfaces[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building_b = context.BASELINE_0
+            return {
+                "scc_dict_b": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building_b
+                ),
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_b = context_item.BASELINE_0
+            scc = data["scc_dict_b"][surface_b["id"]]
+            return (
+                get_opaque_surface_type(surface_b) == OST.ROOF
+                and scc is SCC.UNREGULATED
+            )
+
+        class RoofRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule41.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=produce_ruleset_model_instance(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                    required_fields={
+                        "$": ["optical_properties"],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                roof_b = context.BASELINE_0
+                roof_p = context.PROPOSED
+                scc_dict_b = data["scc_dict_b"]
+                return {
+                    "absorptance_solar_exterior_b": find_one(
+                        "$.optical_properties.absorptance_solar_exterior", roof_b
+                    ),
+                    "absorptance_solar_exterior_p": find_one(
+                        "$.optical_properties.absorptance_solar_exterior", roof_p
+                    ),
+                    "absorptance_thermal_exterior_b": find_one(
+                        "$.optical_properties.absorptance_thermal_exterior", roof_b
+                    ),
+                    "absorptance_thermal_exterior_p": find_one(
+                        "$.optical_properties.absorptance_thermal_exterior", roof_p
+                    ),
+                    "surface_conditioning_category_b": scc_dict_b[roof_b["id"]],
```

Comment:
Remove this `surface_conditioning_category_b`. It is not not needed.

Reply:
Good catch! removed